addon/components/single-event.js

```diff
@@ -99,31 +102,37 @@ export default Component.extend({
     });
   }),
 
-  sessionObjectives: computed('i18n.locale', 'session.sortedObjectives.[]', async function(){
+  sessionObjectives: computed('i18n.locale', 'event.sessionObjectives.[]', 'event.competencies.[]', async function(){
     const i18n = this.get('i18n');
-    const session = await this.get('session');
-    const objectives = await session.get('sortedObjectives');
-    return await map(objectives, async objective => {
-      const parents = await objective.get('topParents');
-      const parent =  parents.get('firstObject');
-      const competency = await parent.get('competency');
+    const event = this.get('event');
+    const objectives =  event.sessionObjectives;
+    const competencies = event.competencies;
+    return objectives.map(objective => {
       //strip all HTML
-      let title = objective.get('title').replace(/(<([^>]+)>)/ig,"");
-      let position = objective.get('position');
-      if(isEmpty(competency)) {
+      const title = objective.title.replace(/(<([^>]+)>)/ig,"");
+      const position = objective.position;
+      if(isEmpty(objective.competencies)) {
         return {
+          id: objective.id,
           title,
           domain: i18n.t('general.noAssociatedCompetencies'),
           position
         };
       }
-      const domain = await competency.get('domain');
+      const competencyId = objective.competencies[0];
+      const competency = competencies.findBy('id', competencyId);
+      const parentId = competency.parent;
+      let domain = competency;
+      if (! isEmpty(parentId)) {
+        domain = competencies.findBy('id', parentId);
+      }
       return {
+        id: objective.id,
         title,
-        domain: competency.get('title') + ' (' + domain.get('title') + ')',
+        domain: competency.title + ' (' + domain.title + ')',
         position
       };
-    });
+    }).sort(this.positionSortingCallback);
```

Comment:
Just curious - how were we sorting this before? I don't see any where that was happening.

Reply:
we're using the output of the `sortedObjectives` CP before, which uses the callback provided by  `sortable-by-position` mixin.